### DISCUSSION TITLE
關於二次開發時報錯

### DISCUSSION THREAD

**Discussion by katelynn620** [theme-shoka-x/hexo-theme-shokaX#84] [Q&A] [answered]
萌新一枚，由於想嘗試自己實作一些功能，所以使用shokaX-CLI選擇從GitHub上安裝以便進行二次開發，不過使用`hexo s`時會出現錯誤，懇請各位前輩多多指點

操作步驟:
1. 使用`hexo init <folder>`新增空白hexo專案
1. 使用`SXC install -r github shokaX`安裝主題
1. 修改`_config.yml`，修改主題為`shokaX`、關閉內建`highlight`以及新增Markdown渲染相關功能
1. 切換至`themes/shokaX`目錄，使用`pnpm build`進行依賴安裝以及編譯TypeScript檔案
1. 使用`hexo s`進行測試，執行沒有問題，也可正常存取`http://localhost:4000`並且看到頁面
1. 將`_config.landscape.yml`更名為`_config.shokaX.yml`，並任意新增一些內容
1. 使用`hexo s`進行測試，出現錯誤並退出

錯誤訊息如以下:
![image](https://user-images.githubusercontent.com/816515/236409875-61084d79-1496-44df-bc7d-a2ae89ce25c5.png)

想請教一下，這個看起來是找不到`hexo-util`這個套件，但上面應該是安裝了啊？還是有遺漏什麼呢？非常感謝🙏

**Comment by zkz098** [comment]
很奇怪的问题，ShokaX目录下的ts文件都删除了吗(Hexo 无法识别ts文件，可能导致import错误的模块)
我在本地无法复现这个问题，不排除是一个Bug

**Comment by D-Sketon** [comment, marked as answer]
可以复现
![image](https://user-images.githubusercontent.com/49871906/236513629-798fff42-461d-4f62-b9f9-f90059047ccb.png)
问题在于不支持`import merge from 'hexo-util'`这种导入，如果使用import默认导入会无法识别
应该为
`import { deepMerge } from 'hexo-util'`
或
`const merge = require('hexo-util')`

**Reply by zkz098** [reply]
理论上来说发行版也应该存在此问题，但是测试过程中并未发现

**Reply by D-Sketon** [reply]
> 理论上来说发行版也应该存在此问题，但是测试过程中并未发现

发行版引用的是node_module下的hexo-util，导出方式为`exports.deepMerge = require('./deep_merge');`
github上引用的是.pnpm下的hexo-util，导出方式为`export { default as deepMerge } from './deep_merge';`
大概问题出在这里吧

**Reply by zkz098** [reply]
> 发行版引用的是node_module下的hexo-util，导出方式为`exports.deepMerge = require('./deep_merge');` github上引用的是.pnpm下的hexo-util，导出方式为`export { default as deepMerge } from './deep_merge';` 大概问题出在这里吧


经典CJS和ESM问题()
已经改了，现在应该修好了
